src/txmempool.cpp

```diff
@@ -957,9 +972,21 @@ size_t CTxMemPool::DynamicMemoryUsage() const {
 void CTxMemPool::RemoveStaged(const setEntries &stage, bool updateDescendants, MemPoolRemovalReason reason) {
     AssertLockHeld(cs);
     UpdateForRemoveFromMempool(stage, updateDescendants);
+    std::set<uint256> txids;
     for (txiter it : stage) {
+        txids.insert(it->GetTx().GetHash());
         removeUnchecked(it, reason);
     }
+    if (pcustomcsview && !txids.empty()) {
+        CCustomCSView view(accountsView());
+        accountsView().ForEachUndo([&](UndoKey const & key, CLazySerialize<CUndo>) {
```

Comment:
Why here don't just use `view.ForEachUndo`?

Reply:
It could i like that when you iterate one view to not change it meanwhile, that's why another would help.